rticonnextdds_connector/rticonnextdds_connector.py

```diff
@@ -307,14 +311,32 @@ def getString(self, index, field_name):
 
 		return _move_native_string(c_value)
 
-	def getDictionary(self, index):
+	def getDictionary(self, index, member_name = None):
 		if type(index) is not int:
 			raise ValueError("index must be an integer")
 		if index < 0:
 			raise ValueError("index must be positive")
-		#Adding 1 to index because the C API was based on Lua where indexes start from 1
+		# Adding 1 to index because the C API was based on Lua where indexes start from 1
 		index = index + 1
-		native_json_str = connector_binding.getJSONSample(self.input.connector.native,tocstring(self.input.name),index)
+		if member_name is None:
+			native_json_str = connector_binding.getJSONSample(
+				self.input.connector.native,
+				tocstring(self.input.name),
+				index)
+		elif not isinstance(member_name, str):
+			raise ValueError("member_name must be a string")
+		else:
+			native_json_str = ctypes.c_char_p()
+			retcode = connector_binding.getJSONMember(
+				self.input.connector.native,
+				tocstring(self.input.name),
+				index,
+				tocstring(member_name),
+				ctypes.byref(native_json_str))
+			_check_retcode(retcode)
+			if retcode == _ReturnCode.no_data:
```

Comment:
Agreed solution is to modify lua_binding.1.0 to check if the member is an unset optional before converting to JSON string and return NO_DATA back up to here.

Reply:
This is now done and in review for the lua_binding.1.0 module